### PR TITLE
Update NERSC docs

### LINKED ISSUE #343: NERSC build instructions to be updated
Cori library workarounds are no longer needed with dynamic linking.

### PR DESCRIPTION
**Pull request by prckent**
Closes #343

### CODE REVIEW

File: manual/installation.tex [hunk 1]
```diff
@@ -686,81 +686,82 @@ \subsection{Installing on NERSC Edison Cray XC30}
 
 \subsection{Installing on NERSC Cori, Haswell Partition, Cray XC40}
 Cori is a Cray XC40 with 16-core Intel "Haswell" nodes
-installed at NERSC. The build settings are similar to eos at OLCF, but
-a workround is needed for the newer Cray OS which does not include a static
-libXml2 library. A NERSC installed version and its dependencies are used:
+installed at NERSC.
 
 \begin{verbatim}
+module swap craype-haswell craype-mic-knl
```
Comment (ye-luo) [inline manual/installation.tex]: I think this only for KNL build.

File: manual/installation.tex [hunk 2]
```diff
@@ -686,81 +686,82 @@ \subsection{Installing on NERSC Edison Cray XC30}
...
+module unload cray-libsci  
+module load boost  
 module load cray-hdf5
 module load cmake
 module load fftw
 export FFTW_HOME=$FFTW_DIR/..
```
Comment (ye-luo) [inline manual/installation.tex]: do we still need fftw? MKL has provided FFT

File: manual/installation.tex [hunk 3]
```diff
@@ -686,81 +686,82 @@ \subsection{Installing on NERSC Edison Cray XC30}
...
-installed at NERSC. The build settings are similar to eos at OLCF, but
-a workround is needed for the newer Cray OS which does not include a static
-libXml2 library. A NERSC installed version and its dependencies are used:
+installed at NERSC.
 
 \begin{verbatim}
+module swap craype-haswell craype-mic-knl
+module unload cray-libsci  
+module load boost  
 module load cray-hdf5
```
Comment (ye-luo) [inline manual/installation.tex]: Does cori has parallel hdf5?

File: manual/installation.tex [hunk 4]
```diff
@@ -686,81 +686,82 @@ \subsection{Installing on NERSC Edison Cray XC30}
...
 
 \subsection{Installing on NERSC Cori, Xeon Phi KNL Knight's Landing partition, Cray XC40}
 The second phase of NERSC's Cori uses Intel
-Xeon Phi Knight's Landing (KNL) nodes. Substantive optimizations for QMCPACK on KNL will
-be forthcoming. The following build recipe ensures that the code
+Xeon Phi Knight's Landing (KNL) nodes. The following build recipe ensures that the code
 generation is appropriate for the KNL nodes:
 
 \begin{verbatim}
+module unload cray-libsci
```
Comment (ye-luo) [inline manual/installation.tex]: is the module swap to knl here?

File: manual/installation.tex [hunk 5]
```diff
@@ -686,81 +686,82 @@ \subsection{Installing on NERSC Edison Cray XC30}
...
+Xeon Phi Knight's Landing (KNL) nodes. The following build recipe ensures that the code
 generation is appropriate for the KNL nodes:
 
 \begin{verbatim}
+module unload cray-libsci  
+module load boost  
 module load cray-hdf5
 module load cmake
 module load fftw
 export FFTW_HOME=$FFTW_DIR/..
```
Comment (ye-luo) [inline manual/installation.tex]: Similar thing for PHDF5 and FFTW

File: manual/installation.tex (hunk 1 above)
Comment (prckent) [reply manual/installation.tex]: cut and paste problem, will fix

File: manual/installation.tex (hunk 2 above)
Comment (prckent) [reply manual/installation.tex]: will check

File: manual/installation.tex (hunk 4 above)
Comment (prckent) [reply manual/installation.tex]: will fix

File: manual/installation.tex (hunk 5 above)
Comment (prckent) [reply manual/installation.tex]: won't check phdf5 in this PR. No time.

Comment (prckent) [reply manual/installation.tex]: Turned out to be simple. Please merge if you are OK with the updated PR.

**Comment by ye-luo** [comment]
Simplicity is what we like.

**Pull request merged**